components/patient-access-rules/migrations/src/test/java/org/phenotips/data/permissions/internal/R71509PhenoTips3476DataMigrationTest.java

```diff
@@ -0,0 +1,213 @@
+        when(xwiki.getDocument(r2, xc)).thenReturn(p2);
+        BaseObject o2 = mock(BaseObject.class);
+        when(p2.newXObject(Visibility.CLASS_REFERENCE, xc)).thenReturn(o2);
+        when(cm.getInstance(XWikiStoreInterface.class,
+            "hibernate")).thenThrow(new ComponentLookupException("Nope"));
+        callback.doInHibernate(session);
+    }
+
+    /** Unreadable data doesn't affect the migrator. */
+    @Test
+    public void hibernateMigrateIgnoresMissingDocuments() throws Exception
+    {
+        this.mocker.registerMockComponent(ComponentManager.class);
+        ComponentManager cm = this.mocker.getInstance(ComponentManager.class);
+        XWikiHibernateStore store = mock(XWikiHibernateStore.class);
+        when(cm.getInstance(XWikiStoreInterface.class, "hibernate")).thenReturn(store);
+        Execution e = this.mocker.getInstance(Execution.class);
+        ExecutionContext ec = mock(ExecutionContext.class);
+        when(e.getContext()).thenReturn(ec);
+        XWikiContext xc = mock(XWikiContext.class);
+        when(ec.getProperty("xwikicontext")).thenReturn(xc);
+        @SuppressWarnings("deprecation")
+        ArgumentCaptor<HibernateCallback<Object>> callbackCaptor = new ArgumentCaptor<>();
+        this.mocker.getComponentUnderTest().migrate();
+        Mockito.verify(store).executeWrite(Matchers.same(xc), callbackCaptor.capture());
+
+        XWiki xwiki = mock(XWiki.class);
+        when(xc.getWiki()).thenReturn(xwiki);
+        HibernateCallback<Object> callback = callbackCaptor.getValue();
+        Assert.assertNotNull(callback);
+        Session session = mock(Session.class);
+        Query q = mock(Query.class);
+        when(session.createQuery(Matchers.anyString())).thenReturn(q);
+        List<String> docs = new ArrayList<>();
+        docs.add("Family.NotFound");
+        when(q.list()).thenReturn(docs);
+        callback.doInHibernate(session);
+    }
+
+    /** Non empty description. */
+    @Test
+    public void getDescription() throws Exception
+    {
+        Assert.assertTrue(StringUtils.isNotBlank(this.mocker.getComponentUnderTest().getDescription()));
+    }
+
+    /** Non empty name. */
+    @Test
+    public void getName() throws Exception
+    {
+        Assert.assertTrue(StringUtils.isNotBlank(this.mocker.getComponentUnderTest().getName()));
+    }
+
+    /** Correct version number. */
+    @Test
+    public void getVersion() throws Exception
+    {
+        Assert.assertEquals(71509, this.mocker.getComponentUnderTest().getVersion().getVersion());
+    }
+
+    /**
+     * Always executes. When the initial database version is greater than this migrator's, the manager decides not to
+     * run.
+     */
+    @Test
+    public void shouldExecute() throws Exception
+    {
+        Assert.assertTrue(this.mocker.getComponentUnderTest().shouldExecute(new XWikiDBVersion(54689)));
+        Assert.assertTrue(this.mocker.getComponentUnderTest().shouldExecute(new XWikiDBVersion(54693)));
```

Comment:
These numbers are copy/pasted, change them to `71500` and `71510`.

Reply:
if this component is version 71509 why should it execute for DB version 71510? Should it be `assertFalse` then?

Reply:
The pruning based on the current version is done someplace else, this method is for a different purpose. It helps with scenarios like:

- this migrator does the same as an older one, so don't run if the other one didn't run yet and will also run in the current batch
- this migrator fixes something broken by another migrator, so only run it the other one didn't run yet

Reply:
so in this particular case why should it run when DB version is 71510?

Reply:
Because it has no explicit conflict with that version, and the migration manager is the one that already decides that 71509 shouldn't run when the db is already at 71510.